compiler/rustc_lint/src/unnecessary_send_constraint.rs

```diff
@@ -0,0 +1,56 @@
+use rustc_span::sym;
+
+use crate::hir;
+
+use crate::{lints::UnnecessarySendConstraintDiag, LateContext, LateLintPass};
+
+declare_lint! {
+    /// The `lint_unnecessary_send_constraint` lints unnecessary constraint of references to `Send`.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,compile_fail
+    /// fn foo(_: &(dyn Any + Send>) {}
+    /// ```
+    ///
+    /// {{produces}}
+    ///
+    /// ### Explanation
+    ///
+    /// References cannot be sent across threads unless they have a `Sync` bound, so constraining them to `Send` without `Sync` is unnecessary.
```

Comment:
This explanation seems wrong/unrelated. This sentence talks about the reference itself being `Sync`/`Send`, but the `Any` and `Send` in `&(dyn Any + Send)` is about the object the reference points to, not about the reference itself. `Send` is useless there because a `&T` does not allow sending the `T`.

Reply:
I think it talks about the object being `Sync`, which is exactly what is needed to make the reference `Send`, which presumably is what the programmer intended to achieve.